flake.nix

```diff
@@ -38,6 +38,9 @@
       nightlyBuild = pkgs.callPackage ./nix/build.nix {
         inherit craneLib rustPlatform;
       };
+
+      # Override default rustc and cargo
+      inherit (craneLib) rustc cargo;
```

Comment:
This is overriding attributes from nixpkgs by attriutes from crane. This is to make `shell.nix` usable without flake (with default global `<nixpkgs>`).